binding-mri/binding-mri.cpp

```diff
@@ -32,7 +32,7 @@
 #include "boost-hash.h"
 
 #include <ruby.h>
-#if RUBY_API_VERSION_MAJOR > 1
+#ifndef RUBY_LEGACY_VERSION
```

Comment:
I define it. Problem is that checking for 1.8 vs. 1.9 is ugly to write so I introduced a new macro for 1.8 only. You can suggest a better macro name :D